ACM Digital Library.js

```diff
@@ -1,203 +1,200 @@
+	"creator": "Guy Aglionby",
+	"target": "^https://dl\\.acm\\.org/(doi|do|profile|toc|topic|keyword|action/doSearch|acmbooks)",
 	"minVersion": "3.0",
 	"maxVersion": "",
 	"priority": 100,
 	"inRepository": true,
 	"translatorType": 4,
 	"browserSupport": "gcsibv",
-	"lastUpdated": "2019-10-03 09:45:46"
+	"lastUpdated": "2019-12-31 14:12:38"
 }
 
 /*
-ACM Digital Library Translator
-Copyright (C) 2011 Sebastian Karcher and CHNM
+	***** BEGIN LICENSE BLOCK *****
 
-This program is free software: you can redistribute it and/or modify
-it under the terms of the GNU General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
+	Copyright © 2019 Guy Aglionby
+	This file is part of Zotero.
 
-This program is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-GNU General Public License for more details.
-
-You should have received a copy of the GNU General Public License
-along with this program. If not, see <http://www.gnu.org/licenses/>.
-*/
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
 
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
 
-// attr()/text() v2
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
 
+	***** END LICENSE BLOCK *****
+*/
 
 function detectWeb(doc, url) {
-	if (url.includes("/results.cfm") || url.includes("/author_page.cfm") || url.includes("/ccs/ccs.cfm")) {
-		return getSearchResults(doc, true) ? 'multiple' : false;
-	} else if (url.includes("/citation.cfm")) {
-		return getArticleType(doc);
+	if ((url.includes('/doi/') || url.includes('/do/')) && !url.includes('/doi/proceedings')) {
+		let extractedContext = doc.querySelector('meta[name=pbContext]').content;
+		let subtypeRegex = /csubtype:string:(\w+)/;
+		let subtype = extractedContext.match(subtypeRegex)[1].toLowerCase();
+
+		if (subtype == 'conference') {
+			return 'conferencePaper';
+		}
+		else if (subtype == 'journal' || subtype == 'magazine' || subtype == 'newsletter') {
```

Comment:
[Here](https://dl.acm.org/doi/10.1145/2854146) is a magazine ([corresponding test](https://github.com/zotero/translators/pull/2101/commits/a853a19e0dfa55c877a5731e71378e362f9f00a1#diff-2b2fb3a06b538f7aa0f0d1f03670c891R558)); [here](https://dl.acm.org/doi/10.1145/3264631.3264634) is a newsletter ([corresponding test](https://github.com/zotero/translators/pull/2101/commits/a853a19e0dfa55c877a5731e71378e362f9f00a1#diff-2b2fb3a06b538f7aa0f0d1f03670c891R518)).

Reply:
So, they don't have an submit area for authors and are maybe otherwise a little different. However, I see that dblp is classify them also as journals. Thus, I am okay with this now.